### PR TITLE
Implement JSDoc

### LINKED ISSUE #69: implement jsdoc or other standardized and automatable doc tool
#68 @falquaddoomi made the comment:

> Looks good. My only suggestion might be using JSDoc-style comments if we ever decide to run a documentation generator on the source and it's not too tedious for you. There are a bunch of options for parsing out the comments and generating documentation (Vuese, styleguidist, plugins for JSDoc), but I haven't tried any of them myself.

This would be helpful as a way to more consistently document components (and other things) in a consistent way. It also might be a good intermediate solution between no dedicated component docs and a full-blown storybook playground (overkill).

It might also allow for better editor integration? I already have prop types working, such that when using a component it will alert you if you're missing required props, or give an invalid value for a prop. It'd be cool if the IDE could also show the documentation along with it (e.g. "this component should be used in this circumstance").

I'd add that whatever the solution, it should be applicable to every vue component as well, even ones without a `<script>` tag. I should perhaps move my one-sentence description of components, which I've already done for most, from right before `defineComponent` to the top of the `<template>` so it's right at the top of the file.

### LINKED ISSUE #134: implement jsdoc or other standardized and automatable doc tool
monarch-initiative/monarch-ui-new#68 @falquaddoomi made the comment:

> Looks good. My only suggestion might be using JSDoc-style comments if we ever decide to run a documentation generator on the source and it's not too tedious for you. There are a bunch of options for parsing out the comments and generating documentation (Vuese, styleguidist, plugins for JSDoc), but I haven't tried any of them myself.

This would be helpful as a way to more consistently document components (and other things) in a consistent way. It also might be a good intermediate solution between no dedicated component docs and a full-blown storybook playground (overkill).

It might also allow for better editor integration? I already have prop types working, such that when using a component it will alert you if you're missing required props, or give an invalid value for a prop. It'd be cool if the IDE could also show the documentation along with it (e.g. "this component should be used in this circumstance").

I'd add that whatever the solution, it should be applicable to every vue component as well, even ones without a `<script>` tag. I should perhaps move my one-sentence description of components, which I've already done for most, from right before `defineComponent` to the top of the `<template>` so it's right at the top of the file.

### PR DESCRIPTION
**Pull request by vincerubinetti**
Closes monarch-initiative/monarch-api#134 

- convert all `// standard inline js comments` to `/** basic jsdoc comments */`
- make note in readme about just always using jsdoc comments for consistency
- use `/* block js comments */` for the few cases where we have some small code we might want later
- get rid of TODO comments, as they're all tracked in github issues
- install prettier jdsoc plugin that lints and wraps long comments
- upgrade packages to support the above

Basically the main point of this PR is that, using JSDoc comments instead of JS, editors (at least VS Code's intellisense and Volar extension) show the adjacent comment when you hover over [practically any entity (with a few small gaps)](https://github.com/johnsoncodehk/volar/discussions/1473). Really I don't know why they don't do the same thing with regular comments though.

Screen Recording 2022-06-17 at 5 42 36 PM

(gif in slow motion for some reason)

### CODE REVIEW

**Review by falquaddoomi — COMMENTED** [review]
Take this with a grain of salt as ultimately you'll be maintaining the codebase, not me, but I personally use JSDoc comments for documenting the interface of the code, not the inner workings. I.e., I wouldn't replace every inline comment with a JSDoc comment, but only the ones that annotate an entity: a function, object, member, class, interface, etc. I wouldn't use a JSDoc to indicate, e.g., why I was doing something in the following code or to describe the flow of execution. Outside of VS Code tooltips, JSDocs are often used to generate static documentation from the source, so using JSDoc comments so often might create very verbose documentation.

If you disagree with my opinion, no worries, I'll approve. I didn't see anything surprising while going through the changes, so I think it's fine as-is.

**Comment by vincerubinetti** [comment]
The reason I chose to convert them all is because 1) it was mostly a simple find and replace operation, 2) I wanted to make it more "foolproof" (like, no question of whether to use normal or jsdoc comments, just always use the same type), 3) it seems like a majority of my comments are before entities, and a smaller portion are "headings" before blocks of code, and 4) we get linting/formating for comments now too through the plugin (prettier itself doesn't support wrapping comments yet).

Regarding 3), this PR also puts comments before local variable declarations within functions/loops/etc. This isn't something I would think to look for, but playing around with it, I found it really nice to be able to hover over a variable that's used further down in a function and see what it is, rather than having to `ctrl+D` back up to it.

Regarding doc generators: I talk about how I don't see why we'd ever need to run doc generators in #69 , that I only wanted jsdoc for intellisense really. I think generated docs would only add more things to maintain and more confusion. If someone in the future did want generated docs, I'm not sure how they would handle comments within a function, for example. Maybe they would omit those anyway?

**Review by falquaddoomi — APPROVED** [review]

**Comment by vincerubinetti** [comment]
Added small commit that 1) ignores msw.js from eslint/prettier so we dont change it every time we do a fresh install and the file gets regenerated (sort of relevant because the comments were being formatted there), 2) makes all "dangling" vue event attributes explicitly `undefined` to fix weird volar issue (sort of relevant because this only started happening with the package upgrades I had to do here), 3) fresh yarn.lock install.

**Comment by vincerubinetti** [comment]
> but I personally use JSDoc comments for documenting the interface of the code, not the inner workings. I.e., I wouldn't replace every inline comment with a JSDoc comment

Just wanted to come back and explicitly say that this was a very valid comment. Doing this PR felt a little bit weird, and certainly off the beaten path of typical JS practice. But the benefits are already adding value for me. Also, we can always go back to normal comments pretty easily with find+replace if we want to; this PR still went through and made things more consistent.
